### PR TITLE
tests: Further pipeline viewport tests cleanup

### PR DESCRIPTION
**Pull request by krOoze**
- using available abstractions and utils to improve the tests' code

### CODE REVIEW

**Comment by jzulauf-lunarg** [comment]
I've got a helper class in review that is designed to replace much of the bolierplate code you are fixing.  Can your review for suitability and gaps? https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/2220

**Comment by krOoze** [comment]
@jzulauf-lunarg Yea, I think I am gonna go that path, now that your PR was accepted.

**Comment by krOoze** [comment]
Updated with the suggestion in mind.

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -9672,15 +9609,13 @@ TEST_F(VkLayerTest, PSOViewportStateTests) {
...
-        vp_state_ci.viewportCount = test_case.viewport_count;
-        vp_state_ci.pViewports = test_case.viewports;
-        vp_state_ci.scissorCount = test_case.scissor_count;
-        vp_state_ci.pScissors = test_case.scissors;
-
-        for (const auto vuid : test_case.vuids) m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, vuid);
-        VkPipeline pipeline;
-        vkCreateGraphicsPipelines(m_device->device(), VK_NULL_HANDLE, 1, &gp_ci, nullptr, &pipeline);
-        m_errorMonitor->VerifyFound();
+        const auto brake_vp = [&test_case](CreatePipelineHelper &helper) {
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: break_vp instead of brake_vp.

Comment (krOoze) [reply tests/layer_validation_tests.cpp]: Eh, potato tomato. Fixed

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -1195,6 +1196,11 @@ struct CreatePipelineHelper {
             test.Monitor()->VerifyFound();
         }
     };
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: This is conflicting with your "055fc8fd4  build: Clear some compiler warnings" -- need to remove ; also below.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -1195,6 +1196,11 @@ struct CreatePipelineHelper {
...
+
+    template <typename Test, typename OverrideFunc, typename Error>
+    static void OneshotTest(Test &test, OverrideFunc &info_override, const VkFlags flags, Error error, bool positive_test = false) {
+        OneshotTest(test, info_override, flags, std::vector<Error>(1, error), positive_test);
+    };
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: Remove ; to be consistent with your changes from  "055fc8fd4 build: Clear some compiler warnings"

File: tests/layer_validation_tests.cpp (hunk 2 above)
Comment (krOoze) [reply tests/layer_validation_tests.cpp]: indeed

File: tests/layer_validation_tests.cpp (hunk 3 above)
Comment (krOoze) [reply tests/layer_validation_tests.cpp]: Rebased.

**Pull request merged**